Sources/Utility/ArgumentParser.swift

```diff
@@ -82,64 +81,61 @@ public protocol ArgumentKind {
     ///
     /// Call methods on the passed parser to manipulate parser as needed.
     init(parser: inout ArgumentParserProtocol) throws
-
-    /// This will be called for positional arguments with the value discovered.
-    init?(arg: String)
-
+    
+    /// Failable initializer to convert from String.
+    init?(argument: String)
```

Comment:
This is just a name change, right? Please propose this separately to avoid the noise in big PR like this.